src/viewport/FlyCameraManipulator.cpp

```diff
@@ -0,0 +1,111 @@
+#include "FlyCameraManipulator.h"
+#include "Commands.h"
+#include "Gui.h"
+#include "Viewport.h"
+#include "Editor.h"
+#include <cmath>
+#include <pxr/base/gf/plane.h>
+#include <pxr/usd/usdGeom/camera.h>
+
+FlyCameraManipulator::FlyCameraManipulator(const GfVec2i &viewportSize, bool isZUp) : CameraRig(viewportSize, isZUp) {}
+
+void FlyCameraManipulator::OnBeginEdition(Viewport &viewport) {
+    if (viewport.IsEditingStageCamera()) {
+        _stageCamera = UsdGeomCamera::Get(viewport.GetCurrentStage(), viewport.GetSelectedStageCameraPath());
+        BeginEdition(viewport.GetCurrentStage());
+    }
+}
+
+void FlyCameraManipulator::OnEndEdition(Viewport &viewport) {
+    if (viewport.IsEditingStageCamera() && _stageCamera) {
+        EndEdition();
+    }
+}
+
+float FlyCameraManipulator::InputForward() {
+    if (ImGui::IsKeyDown(ImGuiKey_W) || ImGui::IsKeyDown(ImGuiKey_UpArrow))
+        return 1.0f;
+    if (ImGui::IsKeyDown(ImGuiKey_S) || ImGui::IsKeyDown(ImGuiKey_DownArrow))
```

Comment:
would be better to have an `else if` instead of ìf`since it allows to avoid testing more conditions and save some cpu work